tests/unit/TickingMockClockTest.php

```diff
@@ -0,0 +1,120 @@
+        $timeFormat = 'Y-m-d\TH:i:s.u';
+        $now = \DateTimeImmutable::createFromFormat($timeFormat, $timeString, new \DateTimeZone('UTC'));
+        \assert($now instanceof \DateTimeImmutable);
+        $clock = TickingMockClock::tickingFromCurrentTime($now);
+
+        TickingTimeAssertions::assertDateTimeLessThanOneSecondAfter(
+            $clock->tickingFrom(),
+            new \DateTimeImmutable('now'),
+        );
+
+        // Immutable variant.
+
+        $sample1 = $clock->now();
+        $sample2 = $clock->now();
+        $sample3 = $clock->now();
+
+        $this->assertNotSame($sample1->format($timeFormat), $sample2->format($timeFormat));
+        $this->assertNotSame($sample1->format($timeFormat), $sample3->format($timeFormat));
+        $this->assertNotSame($sample2->format($timeFormat), $sample3->format($timeFormat));
+
+        $this->assertGreaterThan($sample1, $sample2);
+        $this->assertGreaterThan($sample1, $sample3);
+        $this->assertGreaterThan($sample2, $sample3);
+
+        TickingTimeAssertions::assertDateTimeLessThanOneSecondAfter($now, $sample1);
+        TickingTimeAssertions::assertDateTimeLessThanOneSecondAfter($now, $sample2);
+        TickingTimeAssertions::assertDateTimeLessThanOneSecondAfter($now, $sample3);
+
+        // Mutable variant.
+
+        $sample1 = $clock->nowMutable();
+        $sample2 = $clock->nowMutable();
+        $sample3 = $clock->nowMutable();
+
+        $this->assertNotSame($sample1->format($timeFormat), $sample2->format($timeFormat));
+        $this->assertNotSame($sample1->format($timeFormat), $sample3->format($timeFormat));
+        $this->assertNotSame($sample2->format($timeFormat), $sample3->format($timeFormat));
+
+        $this->assertGreaterThan($sample1, $sample2);
+        $this->assertGreaterThan($sample1, $sample3);
+        $this->assertGreaterThan($sample2, $sample3);
+
+        TickingTimeAssertions::assertDateTimeLessThanOneSecondAfter($now, $sample1);
+        TickingTimeAssertions::assertDateTimeLessThanOneSecondAfter($now, $sample2);
+        TickingTimeAssertions::assertDateTimeLessThanOneSecondAfter($now, $sample3);
+    }
+
+    /**
+     * @test
+     */
+     public function it_changes_the_time(): void
+     {
+         $timeString = '2021-05-05T14:11:49.128311';
+         $timeFormat = 'Y-m-d\TH:i:s.u';
+         $now = \DateTimeImmutable::createFromFormat($timeFormat, $timeString, new \DateTimeZone('UTC'));
+         \assert($now instanceof \DateTimeImmutable);
+         $clock = TickingMockClock::tickingFromCurrentTime($now);
+
+         $updatedTime = $now->modify('+30 minutes');
+
+         $clock->changeTimeTo($updatedTime);
+
+         TickingTimeAssertions::assertDateTimeLessThanOneSecondAfter($updatedTime, $clock->now());
+         TickingTimeAssertions::assertDateTimeLessThanOneSecondAfter($updatedTime, $clock->nowMutable());
+     }
+
+    /**
+     * @test
+     */
+    public function it_returns_a_date_object(): void
```

Comment:
Isn't that explicit from the signature? I wonder if this is similar to the test below but effectively demonstrating that when it has only been ticking from a second or so ago, it hasn't advanced the date?

Reply:
Agree, bad name, it's testing that the date that comes back is the day of the mocked value.

Reply:
```suggestion
    public function it_provides_the_date_from_its_current_time(): void
```